test/framework/toolchain.py

```diff
@@ -178,6 +178,14 @@ def test_prepare_iterate(self):
         tc.prepare()
         self.check_vars_foss_usempi(tc)
 
+    def test_cray_reset(self):
+        init_config(build_options={'optarch': 'test', 'silent': True})
+        tc = self.get_toolchain('CrayGNU', version='2015.06-XC')
+        tc.set_options({'dynamic': True})
+        tc.prepare()
+        tc.reset()
+        tc.prepare()
+
```

Comment:
This set of calls raise an EasyBuildError without the change...@boegel is this enough?

Reply:
That certainly covers it, yeah.

Slighty improved in e2bda0a0f, now good to go imho...

Will merge when Travis gives green light.